### PR TITLE
chore(release): introduce "@ossjs/release"

### PR DESCRIPTION
**Pull request by yshrsmz**
related: #10 

introduce @ossjs/release to automate release flow.

I've tried dry-run and got the following result

```shell
$ release publish --dry-run
INFO: preparing release for "yshrsmz/http-middleware" from branch "ossjs-release"...
INFO: found latest release: v0.3.0 (d217c32d70837ae262546a9bea546103e4c5faa9)
INFO: found 3 new commits:
  - 4b64528474edf08464bb285f3cfbc6c054aa43ab chore: fix permission of .husky/pre-commit
  - 5de309487e044c4c9dd7da29518ebcbb125553c8 chore(release): introduce "@ossjs/release"
  - c241c7badc7059863d5d8e0a04c45a6cae22bf14 feat: update "msw" to 0.40.2 (#10)
INFO: successfully parsed commits: 3
INFO: next release type: minor
INFO: next version: 0.3.0 -> 0.4.0
WARN: skip version bump in package.json in dry-run mode (next: 0.4.0)
INFO: preparing to run the publishing script with:
{"RELEASE_VERSION":"0.4.0"}
WARN: skip executing publishing script in dry-run mode
WARN: skip creating a release commit in dry-run mode: "chore(release): v0.4.0"
WARN: skip creating a release tag in dry-run mode: v0.4.0
WARN: skip pushing release to Git in dry-run mode
INFO: generated release notes:

## v0.4.0 (2022-06-01)

### Features

- update "msw" to 0.40.2 (#10) (c241c7badc7059863d5d8e0a04c45a6cae22bf14)

WARN: skip creating a GitHub release in dry-run mode
WARN: skip commenting on 1 GitHub issue:
  - 10
WARN: release "v0.4.0" completed in dry-run mode!
✨  Done in 2.95s.
```

### CODE REVIEW

**Comment by yshrsmz** [comment]
also, I've fixed the permission of `.husky/pre-commit` because it otherwise throws permission denied error

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          token: ${{ secrets.GH_ADMIN_TOKEN }}
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: release
+        run: yarn release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_ADMIN_TOKEN }}
+          NPM_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
[the remark this answers is not in the document]
Comment (kettanaito) [reply .github/workflows/release.yml:26]: Yes, thanks for copying them. I've set them up.

**Review by kettanaito — APPROVED** [review]
Perfect! Thank you for improving the release process in the repo, @yshrsmz! Welcome to contributors 🎉 

I will trigger the release once we merge it to quickly validate that it works as expected. Also to get those changes you wanted published.

**Comment by kettanaito** [comment]
My bad, I've linked you to the incomplete pull request. We're missing the Git user setup:

https://github.com/mswjs/data/blob/c7fef7b8267e336e1f7f5298e23c984bf1ff6275/.github/workflows/release.yml#L19-L22

@yshrsmz, please, would you have a chance to add this setup as shown above? It would fix the [release errors](https://github.com/mswjs/http-middleware/runs/6707355636?check_suite_focus=true) when it doesn't know who's the author of the release commit.

**Comment by yshrsmz** [comment]
@kettanaito 

Hi, just opened a PR to fix the issue ;)
https://github.com/mswjs/http-middleware/pull/12

**Pull request merged**